### PR TITLE
feat(signer): consolidate signer component and transaction client

### PR DESCRIPTION
**Pull request by snormore**
This PR consolidates the `Signer` component and the `TransactionClient` utility by updating the signer to use the client.

I originally added the client for 2 reasons: (1) to use in tests that submit/execute transactions via a node forwarder/mempool socket, and (2) to use in the committee beacon component for submitting commit/reveal/timeout transactions from the node. But for (2), submitting through a client that's local to the beacon component is still pretty messy in terms of nonce collisions, and the `Signer` component exists for exactly this purpose; a shared component for signing and ordering transactions from the node. The committee beacon use ideally needs to wait for a receipt though, so ultimately it made sense to just refactor the signer to use the client code.

~~The previous signer was buffering the transactions for some time period before sending a batch, but~~ the transaction client just maintains an atomic counter for the next nonce, which gets re-synced at every executed block notification from the notifier (similar to the previous signer implementation). Overall though it's very similar with the added ability to wait for a receipt.

I also included some minor test deflaking with the wait for genesis commits in the pinger and pool components.

### CODE REVIEW

**Comment by matthias-wright** [comment]
> The previous signer was buffering the transactions for some time period before sending a batch
The signer actually doesn't buffer transactions, they are sent to the mempool as they come in: https://github.com/fleek-network/lightning/blob/main/core/signer/src/lib.rs#L150

**Comment by snormore** [comment]
Oh I see, the pending list is for tracking/retrying after initial enqueue. Thanks 🙏

**Comment by matthias-wright** [comment]
I'm trying to figure out what happens when a transaction isn't ordered by consensus.
Looks like the waiting on the receipt will fail after the timeout passes: https://github.com/fleek-network/lightning/blob/c4d733fa50fbf7aec2c94e60d8079022fc9a416b/core/utils/src/transaction/client.rs#L116
This will make call to the `handle` error out: https://github.com/fleek-network/lightning/blob/978043ac19c738ad492564e2c549ab3a4c8cc360/core/signer/src/worker.rs#L68
Is there any retry logic for this case?

**Comment by snormore** [comment]
Hmm no it doesn't retry on timeout atm but I'll update it to, good catch

**Comment by matthias-wright** [comment]
Just to be sure that there isn't a bottleneck: lets say the signer sends txn A and waits for its receipt. If txn B comes in while A is still pending, the signer will simply spawn a new feature that sends txn B and then waits for its receipt. Correct? Basically I wanna make sure that transactions can be send "on the fly" without waiting for the previous transaction to complete. It looks like we always spawn a new [future](https://github.com/fleek-network/lightning/blob/c4d733fa50fbf7aec2c94e60d8079022fc9a416b/core/utils/src/transaction/client.rs#L85) for a new transaction, so that looks good, just wanna make sure there aren't any synchronizations that prevent this.

**Comment by snormore** [comment]
Yep, requests come into `SignerWorker::handle` via the submit socket async, and then inside of `handle` can wait for the receipt. So as long as the [affair logic is to not block](https://github.com/fleek-network/lightning/blob/main/lib/affair/src/lib.rs#L319-L334) on the `handle` futures, we should be good.

**Comment by matthias-wright** [comment]
> The committee beacon use ideally needs to wait for a receipt though

Is it to make sure that the commit and reveal transactions have been ordered by consensus before the timeouts? or is there actual information in the receipt that is needed?

**Comment by snormore** [comment]
Yeah exactly just to make sure it's been ordered/included before moving on, but it's not 100% necessary, it just would be a bit messier without it. There's nothing in the receipt that's needed for it.

**Comment by matthias-wright** [comment]
In the current signer, there was an edge case we had to fix where the retry logic could've produced the same valid transaction twice with different nonces. I just want to make sure we don't have this case here. 
Lets say the signer sends a transaction with nonce n and it times out. Then the signer will try to resend the transaction with nonce n+k. Now there is the same transaction twice with nonce n and nonce n+k. I think the way the atomic nonce works, k will be at least 1. Let say k=1, then I think someone can replay the transaction with nonce n and as long as it is ordered before the transaction with nonce n+k, it will be valid. I could be wrong, but I think this edge case can happen here. What do you think?

**Comment by matthias-wright** [comment]
Yeah, that makes sense to me. The `IncrementNonce` txn is actually how we solved the edge case in the current signer, so we already have this transaction type in our state: https://github.com/fleek-network/lightning/blob/c4d733fa50fbf7aec2c94e60d8079022fc9a416b/core/types/src/transaction.rs#L443

File: core/application/src/tests/epoch_change.rs [hunk 1]
```diff
@@ -122,25 +122,25 @@ async fn test_epoch_change_with_some_non_committee_nodes() {
...
     match result.unwrap_err() {
-        TransactionClientError::Reverted((_, TransactionReceipt { response, .. })) => {
+        ExecuteTransactionError::Reverted((_, TransactionReceipt { response, .. }, _)) => {
             assert_eq!(
                 response,
                 TransactionResponse::Revert(ExecutionError::NotCommitteeMember)
             )
         },
-        _ => panic!("unexpected error type"),
+        e => panic!("unexpected error type: {:?}", e),
```
Comment (ozwaldorf) [inline core/application/src/tests/epoch_change.rs:131]: Teeny tiny nit, can inline all these panics with error printed

clippy lints this for prints so I'd assume they will for panics at some point anyway 

```suggestion
        e => panic!("unexpected error type: {e:?}"),
```

**Comment by matthias-wright** [comment]
are you planning on adding the stuff you mentioned about sending the increment nonce txn before retrying?

**Comment by snormore** [comment]
Yep I added it in https://github.com/fleek-network/lightning/pull/45/commits/9148e27e4522dca0e86a9b47c29a9aa268d1bbac (I should've posted a comment when I pushed it)

**Comment by snormore** [comment]
Thanks for the great review on this 🙏

**Pull request merged**